app/utils/template.ts

```diff
@@ -0,0 +1,178 @@
+/**
+ * Returns a YAML field if the values provided are not empty.
+ *
+ * @example stringArrayToYAMLField("field", ["a", "b"])
+ * // field:
+ * //   - a
+ * //   - b
+ */
+export function stringArrayToYAMLField(
+  fieldName: string,
+  fieldValues: string[],
+  indentLevel = 0
+) {
+  const filteredValues = fieldValues.filter((v) => v !== "");
+
+  if (filteredValues.length === 0) {
+    return "";
+  }
+
+  let indent = "";
+  for (let i = 0; i < indentLevel; i++) indent += "  ";
+  return `\n${indent}${fieldName}:${filteredValues
+    .map((value) => `\n  ${indent}- ${value}`)
+    .join("")}`;
+}
+
+export type ProjectTemplateFields = {
+  name: string;
+  repoUrl: string;
+  description: string;
+  maintainer: string;
+  created: string;
+  languages?: string[];
+  tags?: string[];
+  currentlySeeking?: string[];
+  websiteUrl?: string;
+  twitterUrl?: string;
+  avatar?: string;
+  contributionOverview?: {
+    mainLocation?: string;
+    idealEffort?: string;
+    isMentorshipAvailable?: boolean;
+    automatedDevEnvironment?: string;
+    extras?: string[];
+  };
+  featuredMap?: {
+    url: string;
+    description?: string;
+  };
+  maps?: {
+    url: string;
+    description?: string;
+    subTitle?: string;
+  }[];
+  learnLinks?: { title: string; url: string }[];
+  contributing?: string;
+  overview?: string;
+};
+
+/**
+ * Generates a MDX-compliant string in the format similar to _template.mdx. This
+ * is essentially the model of our projects.
+ */
+export function getTemplateContent(fields: ProjectTemplateFields) {
```

Comment:
Once the form gets validated, we'll assemble the contents of a project into a MDX file, which is made up of YAML at the top and React + Markdown at the bottom. This looks like [the _template.mdx file](https://raw.githubusercontent.com/Codesee-io/opensourcehub/main/public/projects/_template.mdx), if you're curious.